smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -188,6 +188,40 @@ contract MixinTransfer is
     return fee.mul(transferFeeBasisPoints) / BASIS_POINTS_DEN;
   }
 
+  /**
+  * @notice Modify the expirationTimestamp of a key
+  * by a given amount.
+  * @param _owner The owner of the key to modify
+  * @param _deltaT The amount of time in seconds by which
+  * to modify the keys expirationTimestamp
+  * @param _addTime Choose whether to increase or decrease
+  * expirationTimestamp (false == decrease, true == increase)
+  * @dev Throws if owner does not have a valid key.
+  */
+  function _timeMachine(
+    address _owner,
+    uint256 _deltaT,
+    bool _addTime
+  ) internal
+    hasValidKey(_owner)
+  {
+    Key storage key = keyByOwner[_owner];
+    uint formerTimestamp = key.expirationTimestamp;
+    if(_addTime) {
+      if(formerTimestamp.add(_deltaT) > block.timestamp.add(expirationDuration)) {
+        key.expirationTimestamp = block.timestamp.add(expirationDuration);
+      } else {
+        key.expirationTimestamp = formerTimestamp.add(_deltaT);
+      }
+    } else {
+      if(formerTimestamp.sub(_deltaT) <= block.timestamp) {
+        key.expirationTimestamp = block.timestamp;
```

Comment:
This may be okay as either way the key is expired, but is this what we are looking for?

For the partial transfer scenario, if this is < block.timestamp then they may be transferring too much.

Reply:
Agreed. My thinking was that the `partialTransfer()` function (not yet introduced) will need to make the appropriate checks to ensure that a key owner can't transfer more than they have. 
This timeMachine function doesn't/shouldn't have to anticipate all use cases, only that it does what it is supposed to do, add or subtract correctly. I currently see at least 2 different uses for it, paying the transfer fee in time, and partial transfers. They may well be more down the road and each will have different requirements. Does this seem reasonable?

Reply:
Yes it does not need to consider everything.  I'm asking about `_addTime = true`

Use case: I want to add x minutes to account y.  It seems that could be useful for various.  It seems we would want that to work instead of fail because account y does not already own a valid key.  But I can understand wanting to keep new key creation logic separate.

Reply:
lol sorry confusing issues.  this is addTime = false.

Reply:
Okay so for this one - it seems like a silent fail.  Why would we ever call this with a value which is too large?

Reply:
Another way to look at it is why is this helpful?  Isn't deleting these lines valid as well?  If so, let's simplify right?

Reply:
Gotcha. Just as for adding time, when deducting time, the check on how much time can be deducted could/should be made in the calling function, not here. I think this is what you are saying. To simplify, this can just be:
```
else {
      key.expirationTimestamp = formerTimestamp - _deltaT;
    }
```